bundles/binding/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/KM200Comm.java

```diff
@@ -368,12 +379,12 @@ public void initObjects(String service) {
 
                 case "floatValue": /* Check whether the type is a single value containing a float value */
                     logger.debug("initDevice: type float value: {}", decodedData);
-                    valObject = new Float(nodeRoot.getDouble("value"));
+                    valObject = Float.parseFloat(nodeRoot.get("value").toString());
                     newObject.setValue(valObject);
                     if (nodeRoot.has("minValue") && nodeRoot.has("maxValue")) {
                         List<Float> valParas = new ArrayList<Float>();
-                        valParas.add(new Float(nodeRoot.getDouble("minValue")));
-                        valParas.add(new Float(nodeRoot.getDouble("maxValue")));
+                        valParas.add(Float.parseFloat(nodeRoot.get("minValue").toString()));
+                        valParas.add(Float.parseFloat(nodeRoot.get("maxValue").toString()));
```

Comment:
`BigDecimal` is usually a much better solution than `Float` or `Double`, as the exact value can be represented.  Please use `BigDecimal` exclusively going between the KMXXX and item states.